#!/usr/bin/env python
PACKAGE = "geom_rcnn"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("xmin", double_t, 0, "x min", 0, -3, 1)
gen.add("xmax", double_t, 0, "x max", 1.7, 1, 3)
gen.add("ymin", double_t, 0, "y min", -0.6, -1, 0)
gen.add("ymax", double_t, 0, "y max", 0.6, 0, 1)
gen.add("zmin", double_t, 0, "z min", -0.15, -1, 0.5)
gen.add("zmax", double_t, 0, "z max", 1.8, 0.5, 3)

gen.add("euc_cluster_tolerance", double_t, 0, "Euclidean Cluster Tolerance", 0.05, 0.01, 0.8)
gen.add("euc_cluster_min", double_t, 0, "Euclidean Cluster Min", 30, 10, 100)
gen.add("euc_cluster_max", double_t, 0, "Euclidean Cluster Max", 10000, 5000, 50000)

gen.add("concave_hull_alpha", double_t, 0, "Concave Hull Alpha", 0.2, 0.05, 1.0)

gen.add("segmentation_distance_thresh", double_t, 0, "Segmentation Distance Threshold", 0.07, 0.001, 0.3)

exit(gen.generate(PACKAGE, "geom_rcnn", "GeomVals"))


